components/codec/src/number.rs

```diff
@@ -909,15 +958,56 @@ pub trait BufferNumberEncoder: BufferWriter {
         unsafe { self.advance_mut(encoded_bytes) };
         Ok(encoded_bytes)
     }
+
+    /// Writes all unsigned 8 bit integers ,
+    ///
+    /// # Errors
+    ///
+    /// Returns `Error::BufferTooSmall` if buffer remaining size < values.len().
+    #[inline]
+    fn write_all_bytes(&mut self, values: &[u8]) -> Result<()> {
```

Comment:
Could we use `MemComparableByteCodec::encode_all` instead? what's the difference?

Reply:
`MemComparableByteCodec::encode_all` is for memory comparable, it will add some padding. this method just simple put all u8 to the buffer, like the `write_all` methods of `Vec`